src/Octoshift/GithubClient.cs

```diff
@@ -82,7 +86,14 @@ public virtual async IAsyncEnumerable<JToken> GetAllAsync(string url)
             var content = await response.Content.ReadAsStringAsync();
             _log.LogVerbose($"RESPONSE ({response.StatusCode}): {content}");
 
-            response.EnsureSuccessStatusCode();
+            if (status == HttpStatusCode.OK)
+            {
+                response.EnsureSuccessStatusCode();
+            }
```

Comment:
This code is semantically wrong. It will never raise any error at all. I still think the passing a flag like before to not raise an error in case of non success status is a better choice. And I have seen it in other popular projects too, it's a common practice.

Reply:
Oh wait sorry, I got confused this code is OK, as is. I confused it with `response.status == HttpStatusCode.OK`

Reply:
I don't think it's semantically wrong, if you call SendAsync and don't pass in the status param, then EnsureSuccessStatusCode will be run. If you do pass in a status (presumably something other than OK), then it won't check for success in the Client.

Reply:
@dylan-smith yeah it was my bad, the code is ok.